localstack-core/localstack/services/cloudwatch/provider_v2.py

```diff
@@ -889,7 +970,7 @@ def create_message_response_update_state_lambda(
     return json.dumps(response, cls=JSONEncoder)
 
 
-def create_message_response_update_state_sns(alarm, old_state):
+def create_message_response_update_state_sns(alarm: LocalStackMetricAlarm, old_state):
```

Comment:
Good stuff, typing is great!

It looks like these states are `StateValue` instances

```suggestion
def create_message_response_update_state_sns(alarm: LocalStackMetricAlarm, old_state: StateValue):
```

Reply:
Good catch, will add it here as per suggestion and to a new `create_message_response_update_composite_alarm_state_sns` one as well. 👍